cpp/src/io/parquet/writer_impl.cu

```diff
@@ -2472,10 +2426,10 @@ void writer::impl::write_parquet_data_to_sink(
     auto const h_pages = cudf::detail::make_host_vector_sync(pages, _stream);
 
     // add column and offset indexes to metadata
-    for (auto b = 0, r = 0; b < static_cast<size_type>(batch_list.size()); b++) {
-      auto const rnext   = r + batch_list[b];
-      auto curr_page_idx = chunks[r][0].first_page;
-      for (; r < rnext; r++) {
+    if (num_rowgroups != 0) {
+      auto const rnext   = num_rowgroups;
+      auto curr_page_idx = chunks[0][0].first_page;
+      for (auto r = 0; r < rnext; r++) {
```

Comment:
```suggestion
      auto curr_page_idx = chunks[0][0].first_page;
      for (auto r = 0; r < num_rowgroups; r++) {
```

Reply:
Thanks for the suggestions. I tried to remove all duplicate variables but looks like `rnext` slipped under my radar. Let me commit all these suggestions.

Reply:
Committed changes. Thanks again!